x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -841,7 +841,7 @@ func storeAgentInfo(s saver, reader io.Reader) error {
 	if err := fileLock.TryLock(); err != nil {
 		return err
 	}
-	defer fileLock.Unlock()
+	defer fileLock.Unlock() //nolint:errcheck // defered call
```

Comment:
Maybe we can handle the error here, for example with a closure:
```suggestion
	defer func(fileLock *filelock.AppLocker) {
		unlockErr := fileLock.Unlock()
		if unlockErr != nil {
			// log error here or join it with the internal error returned
		}
	}(fileLock)

```